src/data/booksDataSource.ts

```diff
@@ -0,0 +1,50 @@
+import { Book } from "../models";
+import * as fs from "fs";
+import * as path from "path";
+
+// Path to JSON file where books are stored
+const booksDataPath = path.join(__dirname, "booksData.json");
+
+export class BooksDataSource {
```

Comment:
DataSources should also have their own folder, somewhere next to models and resolvers, while keeping `data` only for the JSON files with the actual data.

Reply:
I separated them, the JSON files remained in the data folder, while the data sources where moved to a separate folder called dataSources. All paths have been fixed to point to the right files.